.github/workflows/test_fast.yml

```diff
@@ -130,7 +128,8 @@ jobs:
         run: etc/bin/shellchecker
 
       - name: MyPy
-        run: mypy
+        # specify --python-version to avoid errors regarding site-packages
+        run: mypy --python-version 3.8
```

Comment:
3.7?

Reply:
```
$ mypy --python-version=3.7
[...]/python3.12/site-packages/matplotlib/pyplot.py:246: error: Assignment expressions are only supported in Python 3.8 and greater  [syntax]
Found 1 error in 1 file (errors prevented further checking)
```

I suppose to be safe we will have to split off mypy into its own job 😒

Reply:
> Assignment expressions are only supported in Python 3.8 and greater

I think mypy has reported a legit error here that the tests failed to spot! Thankfully this issue only applies to the developer main loop plugins so is of little consequence. Would introducing an upper pin on matplotlib clear the error?

Reply:
This version of matplotlib would not install on Python 3.7 so there's no error here. It's just that Mypy can't work in Python 3.7 mode when it comes across 3.7-incompatible code in 3rd-party libraries: https://github.com/python/mypy/issues/9972

Reply:
See also https://github.com/cylc/cylc-flow/issues/5736

Reply:
Wait I just realised that since we moved to linting in its own job on GH Actions, we're not installing matplotlib during that job so this error is not raised anymore!

Reply:
Ach dammit, we should probably install the `all` group, one for another day...